_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+          rowCount={dataCount}
+          rows={dataList}
+          columns={columns}
+          onPageChange={onPageChange}
+        ></CompanyTable>
+      )}
+    </>
+  );
+};
+
+export default CompanyTableContainer;
+```
+{% endraw %}
+
+Add `src/app/HomePage.tsx` for the homepage:
+
+{% raw %}
+__HomePage.tsx__
+```tsx
+"use client";
+
+import CompanyTableContainer from "@/components/company/CompanyTableContainer";
+import { Box, Typography } from "@mui/material";
+import { useSearchParams } from "next/navigation";
+
+const HomePage = () => {
+  const searchParams = useSearchParams();
+  const page = searchParams.get("page")
+    ? parseInt(searchParams.get("page") as string)
+    : 1;
+
+  return (
+    <>
+      <Box>
+        <Typography variant="h4" component="h1">
+          Companies
+        </Typography>
+      </Box>
+      <Box mt={2}>
+        <CompanyTableContainer page={page}></CompanyTableContainer>
+      </Box>
+    </>
+  );
+};
+
+export default HomePage;
+```
+{% endraw %}
+
+Replace the contents of `src/app/page.tsx` and change it to render the `HomePage` component:
+
+{% raw %}
+__app/page.tsx__
+```tsx
+import HomePage from "./HomePage";
+
+const Page = () => {
+  return (
+    <HomePage></HomePage>
+  );
+}
+
+export default Page;
+```
+{% endraw %}
+
+Add a component defining the page width, for using it in the root layout. Create `src/layout/WideLayout.tsx` with the following content:
+
+{% raw %}
+__WideLayout.tsx__
+```tsx
+"use client";
+
+import { Container, ThemeProvider, createTheme } from "@mui/material";
+
+const theme = createTheme({
+  typography: {
+    fontFamily: "inherit",
+  },
+});
+
+const WideLayout = (props: { children: React.ReactNode }) => {
+  return (
+    <ThemeProvider theme={theme}>
+      <Container maxWidth="lg" sx={{ mt: 4 }}>
+        {props.children}
+      </Container>
+    </ThemeProvider>
+  );
+};
+
+export default WideLayout;
+```
+{% endraw %}
+
+With the implementation above, the page content will be wrapped in a `ThemeProvider` component, so https://github.com/vercel/next.js/discussions/45433[MUI child components inherit the font family] from the root layout.
+Update the contents of `src/app/layout.tsx` to be:
+
+{% raw %}
+__app/layout.tsx__
+```tsx
+import WideLayout from "@/layout/WideLayout";
+import { Ubuntu} from "next/font/google";
+
+const font = Ubuntu({
+  subsets: ['latin'],
+  weight: ['300','400','500','700'],
+});
+
+export const metadata = {
+  title: "Create Next App",
+  description: "Generated by create next app",
+};
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return (
+    <html lang="en">
+      <body className={font.className}>
+        <WideLayout>{children}</WideLayout>
+      </body>
+    </html>
+  );
+}
+```
+{% endraw %}
+
+Also, remove `src/app/global.css` and `src/app/page.module.css`. Then run the client application with:
+
+```shell
+npm run dev
+```
```

Comment:
I get an error about Axios at this point.

<img width="985" alt="Screenshot 2023-09-08 at 8 47 30 AM" src="https://github.com/oktadev/okta-blog/assets/17892/70ce0524-5111-4eb6-98eb-e5a739e5aed7">

Reply:
I had a bad cache, added the missing import earlier in the post.